nixos/modules/services/networking/ejabberd.nix

```diff
@@ -11,10 +11,16 @@ let
     ${cfg.ctlConfig}
   '';
 
-  ectl = ''${cfg.package}/bin/ejabberdctl ${optionalString (cfg.configFile != null) "--config ${cfg.configFile}"} --ctl-config "${ctlcfg}" --spool "${cfg.spoolDir}" --logs "${cfg.logsDir}"'';
+  settingsFormat = pkgs.formats.yaml {};
+  configFile = settingsFormat.generate "config.yaml" cfg.settings;
 
-  dumps = lib.escapeShellArgs cfg.loadDumps;
+  ectl = ''${cfg.package}/bin/ejabberdctl --config "${
+      if (cfg.configFile != null)
```

Comment:
Why not just set the default value of `configFile` to `settingsFormat.generate "config.yaml" cfg.settings`? Also, `configFile` is kind of redundant with `settings` anyway

Reply:
Thanks! Good point! I changed the default. I would keep `configFile` around, just for backward compatibility. Maybe we can add a deprecation notice and remove it some time in the future, but I don't see much harm in keeping it around.

Reply:
> Thanks! Good point! I changed the default. I would keep `configFile` around, just for backward compatibility. Maybe we can add a deprecation notice and remove it some time in the future, but I don't see much harm in keeping it around.

The way I see it, is that we can add a warning that deprecates `configFile` and says it would be removed in 25.05 (the next unstable version) or something like that, but keeping both of them around is definitely redundant